src/main/cli/cli.c

```diff
@@ -6783,31 +6790,65 @@ void cliProcess(void)
 
```

Comment:
(line above) - you don't want to flush if in noninteractive mode

Reply:
Adjusted

Reply:
Sorry, my mistake. I was reading the comment and it seemed that input buffer is flushed. But actually output is flushed, which does nothing (output buffer should be empty at this point).
The comment or code is wrong (or both)

Reply:
I believe it's echo'd MSP inbound characters that may have arrived after the #

Reply:
Incoming characters would normally cancel CLI request - there must be 100ms delay after non-msp command.

But this does flush echo/prompt characters (and maybe command reply), but a bit late - why wait for next call?

Flushing after `while (serialRxBytesWaiting())` is IMO cleaner.

Reply:
It may have been there prior to the guard time being added and wasn't cleaned up. It solves nothing for the user, as it would still result in garbage in the terminal if there was anything anyway.

Have removed it in favour of a flush after serial processing as it seems a more reasonable outcome.